pkg/sql/sem/types/types.go

```diff
@@ -117,6 +117,35 @@ var (
 	FamPlaceholder T = TPlaceholder{}
 )
 
+// EqualTypes returns true if types x and y are equal. We need this function because
+// the Go == operator is not defined over all T instances, in particular TTuple.
+func EqualTypes(x, y T) bool {
```

Comment:
It seems confusing to have both an `Equivalent` method and an `EqualTypes` method. If I were to see the two methods, I would not know which to use, or I might assume that one was a wrapper for the other. Should we be updating the naming (e.g. `IdenticalTypes`), or maybe adding flags to control the equivalence semantics? Regardless, how come we wouldn't have a method on the `types.T` interface itself rather than a separate function?